pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -118,7 +118,7 @@ func ConvertService(svc coreV1.Service, domainSuffix string, clusterID string) *
 
 	switch svc.Spec.Type {
 	case coreV1.ServiceTypeNodePort:
-		if _, ok := svc.Annotations[NodeSelectorAnnotation]; ok {
+		if _, ok := svc.Annotations[NodeSelectorAnnotation]; !ok {
```

Comment:
@rshriram I think you added this. Was it just inverted originally?

Reply:
hmm.. this was a dumb mistake! and our internal code didnt catch it either.. thanks for spotting this!